examples-gallery/plot_Mississippi_steamboat.py

```diff
@@ -0,0 +1,445 @@
+# %%
+"""
+Mississippi Steamboat
+=====================
+
+A boat is modeled as a rectangular plate with length :math:`a_S` and
+width :math:`b_S`.
+It has a mass :math:`m_S` and is modeled as a rigid body.
+Water wheels are attached to the boat on the left and right side.
+The wheels have radius :math:`r_W` and mass :math:`m_W` and are modeled as
+rigid bodies.
+By running the wheels at different speeds, the boat can be steered.
+The water speed is assumed to be zero. Gravity, in the negative Z direction, is
+unimportant here, hence disregarded.
+
+**Constants**
+
+- :math:`m_S`: mass of the steamboat [kg]
+- :math:`m_W`: mass of the wheels [kg]
+- :math:`r_W`: radius of the wheel [m]
+- :math:`a_S`: length of the steamboat [m]
+- :math:`b_S`: width of the steamboat [m]
+- :math:`c_W`: drag coefficient at wheels [kg/m^2]
+- :math:`c_S`: drag coefficient at steamboat [kg/m^2]
+
+
+**States**
+
+- :math:`x`: X - position of the center of the steamboat [m]
+- :math:`y`: Y - position of the center of the steamboat [m]
+- :math:`q_S`: angle of the steamboat [rad]
+- :math:`q_{LW}`: angle of the left wheel [rad]
+- :math:`q_{RW}`: angle of the right wheel [rad]
+- :math:`u_x`: speed of the steamboat in X direction [m/s]
+- :math:`u_y`: speed of the steamboat in Y direction [m/s]
+- :math:`u_S`: angular speed of the steamboat [rad/s]
+- :math:`u_{LW}`: angular speed of the left wheel [rad/s]
+- :math:`u_{RW}`: angular speed of the right wheel [rad/s]
+
+**Specifieds**
+
+- :math:`t_{LW}`: torque applied to the left wheel [Nm]
+- :math:`t_{RW}`: torque applied to the right wheel [Nm]
+
+"""
+import numpy as np
+import sympy as sm
+from opty.utils import parse_free
+from scipy.interpolate import CubicSpline
+import sympy.physics.mechanics as me
+from opty.direct_collocation import Problem
+import matplotlib.pyplot as plt
+from matplotlib.animation import FuncAnimation
+from matplotlib.patches import Rectangle
+
+
+# %%
+# Set up the Equations of Motion.
+# -------------------------------
+#
+# Set up the geometry of the system.
+#
+# - :math:`N`: inertial frame of reference
+# - :math:`O`: origin of the inertial frame of reference
+# - :math:`A_S`: body fixed frame of the steamboat
+# - :math:`A_{LW}`: body fixed frame of the left wheel
+# - :math:`A_{RW}`: body fixed frame of the right wheel
+# - :math:`Dmc_S`: center of the steamboat
```

Comment:
Is the "mc" supposed to be a subscript?

Reply:
> Is the "mc" supposed to be a subscript?

No, Sir. I always use, if possible Dmc to be the ceter of mass of some body. I must have copied this from some example a long time ago. If I have Dmc in a name in my code I know it is a center of mass of something.

Reply:
I'm asking more because the math renders awkwardly, not because I don't like your naming conventions. You have these symbols as proper math, so I would expect them to have proper super and subscripts.

Reply:
> I'm asking more because the math renders awkwardly, not because I don't like your naming conventions. You have these symbols as proper math, so I would expect them to have proper super and subscripts.

If I understand you correctly, something like $Dmc_S$ looks awkward?

Reply:
I would expect something more like:

$$
D_{{mc}_s}
$$

or

$$
D^{mc}_s
$$

Reply:
We typically use `A_o` or `A^*` for the mass center of rigid body $A$.

Reply:
> We typically use A o or A ∗ for the mass center of rigid body A .

I thought capital A was 'reserved' for body fixed frames?
So, I could change to $D_{S}^{mc}$

Reply:
You can use any letter you want for a frame, I meant that if you have a frame $X$, then the mass center is $X_o$ of $X^*$.

Reply:
> You can use any letter you want for a frame, I meant that if you have a frame X , then the mass center is X o or X ∗ .

So, if the body fixed frame of my steamboat is AS, then $AS_o$ could be its mass center?